pkg/resources/kafka/kafka.go

```diff
@@ -283,11 +284,18 @@ func (r *Reconciler) Reconcile(log logr.Logger) error {
 		if err = r.updateStatusWithDockerImageAndVersion(broker.Id, brokerConfig, log); err != nil {
 			return err
 		}
+		// If dynamic configuration gets an error then let the loop continue to the next broker
+		// combined error will be returned after the broker loop. This solve that case when other brokers could get healthy,
+		// but the loop exits too soon because brokers are unreachable and dynamic configs cannot be set.
 		if err = r.reconcilePerBrokerDynamicConfig(broker.Id, brokerConfig, configMap, log); err != nil {
-			return err
+			perBrokerDynamicConfigCombineError = errors.Combine(perBrokerDynamicConfigCombineError, err)
```

Comment:
We could add some details for the error like `broker.Id, brokerConfig, configMap`